client/src/featureform/sqlite_metadata.py

```diff
@@ -300,96 +342,154 @@ def fetch_data_safe(self, query, type, name, variant):
+        query = '''SELECT v.*, t.tag_list as tags, p.property_list as properties
+        FROM label_variant v
+        LEFT JOIN tags t ON t.name = v.name AND t.variant = v.variant
+        LEFT JOIN properties p ON p.name = v.name AND p.variant = v.variant
+        WHERE v.source_name = '{0}' AND v.source_variant = '{1}';
+        '''.format(name, variant)
         return self.fetch_data_safe(query, "label_variant", name, variant)
 
     def get_source_variant(self, name, variant):
-        query = f"SELECT * FROM source_variant WHERE name = '{name}' AND variant = '{variant}';"
+        query = '''SELECT v.*, t.tag_list as tags, p.property_list as properties
+        FROM source_variant v
+        LEFT JOIN tags t ON t.name = v.name AND t.variant = v.variant
+        LEFT JOIN properties p ON p.name = v.name AND p.variant = v.variant
+        WHERE v.name = '{0}' AND v.variant = '{1}';
+        '''.format(name, variant)
         return self.fetch_data(query, "source_variant", name, variant)[0]
 
     def get_source_variants_from_source(self, name):
-        return self.query_resource("source_variant", "name", name)
+        return self.query_resource_variant("source_variant", "name", name)
 
     def get_training_set_from_features(self, name, variant):
-        query = f"SELECT * FROM training_set_features WHERE feature_name = '{name}' AND feature_variant = '{variant}';"
+        query = '''SELECT v.*, t.tag_list as tags, p.property_list as properties
+        FROM training_set_features v
+        LEFT JOIN tags t ON t.name = v.training_set_name AND t.variant = v.training_set_variant
+        LEFT JOIN properties p ON p.name = v.training_set_name AND p.variant = v.training_set_variant
+        WHERE v.feature_name = '{0}' AND v.feature_variant = '{1}';
+        '''.format(name, variant)
         return self.fetch_data_safe(query, "training_set_features", name, variant)
 
     def get_training_set_from_labels(self, name, variant):
-        query = f"SELECT * FROM training_set_variant WHERE label_name = '{name}' AND label_variant = '{variant}';"
+        query = '''SELECT v.*, t.tag_list as tags, p.property_list as properties
+        FROM training_set_variant v
+        LEFT JOIN tags t ON t.name = v.name AND t.variant = v.variant
+        LEFT JOIN properties p ON p.name = v.name AND p.variant = v.variant
+        WHERE v.label_name = '{0}' AND v.label_variant = '{1}';
+        '''.format(name, variant)
         return self.fetch_data_safe(query, "training_set_variant", name, variant)
 
     def get_training_set_features(self, name, variant):
-        query = f"SELECT * FROM training_set_features WHERE training_set_name = '{name}' AND training_set_variant = '{variant}'"
+        query = '''SELECT v.*, t.tag_list as tags, p.property_list as properties
+        FROM training_set_features v
+        LEFT JOIN tags t ON t.name = v.training_set_name AND t.variant = v.training_set_variant
```

Comment:
just a thought, we could separate out the join str and the base string

Reply:
You mean into constants to reduce repetition?

Reply:
yeahhh, the motivation was all the repitition

Reply:
Not all these join conditions are the same, but you're right that I could save some repetition. Let me circle back around to this so I can tackle the hosted implementation first. I'll leave this comment unresolved in the meantime.